CMakeLists.txt

```diff
@@ -27,7 +28,17 @@ aux_source_directory(${CMAKE_CURRENT_LIST_DIR}/src/internal ALL_SRC)
 file(GLOB ALL_HEADERS ${CMAKE_CURRENT_LIST_DIR}/src/*.h )
 set(SIO_INCLUDEDIR ${CMAKE_CURRENT_LIST_DIR})
 
+if(EMBED_CACERTS)
+enable_language(ASM)
+set(ALL_SRC ${ALL_SRC} "${CMAKE_CURRENT_LIST_DIR}/src/ca_certificates.s")
+endif()
+
 add_library(sioclient ${ALL_SRC})
+
+if(EMBED_CACERTS)
+target_compile_definitions(sioclient PRIVATE -DEMBED_CACERTS)
```

Comment:
would be good to prefix with SIO_, also, -D not useful

Reply:
Fixed